discovery-provider/src/tasks/index_aggregate_tips.py

```diff
@@ -21,26 +28,63 @@
         , amount
     )
     SELECT 
-            sender_user_id
-            , receiver_user_id
-            , SUM(amount) as amount
+        sender_user_id
+        , receiver_user_id
+        , SUM(amount) as amount
     FROM user_tips
     WHERE
-            slot > :prev_slot AND
-            slot <= :current_slot
+        slot > :prev_slot AND
+        slot <= :current_slot
     GROUP BY
-            sender_user_id
-            , receiver_user_id
+        sender_user_id
+        , receiver_user_id
     ON CONFLICT (sender_user_id, receiver_user_id)
     DO UPDATE
         SET amount = EXCLUDED.amount + aggregate_user_tips.amount
 """
 
 
-def _update_aggregate_tips(session: Session):
+GET_AGGREGATE_USER_TIPS_RANKS_QUERY = """
+SELECT 
+    rank
+    , sender_user_id
+    , receiver_user_id
+    , amount
+FROM (
+    SELECT
+        RANK() OVER (PARTITION BY receiver_user_id ORDER BY amount DESC) AS rank
+        , sender_user_id
+        , receiver_user_id
+        , amount
+    FROM aggregate_user_tips
+    WHERE receiver_user_id IN (
+        SELECT DISTINCT ON (receiver_user_id) receiver_user_id
+        FROM user_tips
+        WHERE
+            slot > :prev_slot AND
+            slot <= :current_slot
+    )
+) rankings
+WHERE rank <= :leaderboard_size
+"""
 
+
+def _get_ranks(session: Session, prev_slot: int, current_slot: int):
+    return session.execute(
+        text(
+            GET_AGGREGATE_USER_TIPS_RANKS_QUERY,
+        ),
+        {"prev_slot": prev_slot, "current_slot": current_slot, "leaderboard_size": 5},
```

Comment:
TODO: Move `5` into a constant somewhere or make configurable?

Alternatively, could choose a large number here (maybe 100 for starters) and have identity choose what to ignore so we can quickly configure the notifs